docs/user/ppl/cmd/patterns.rst

```diff
@@ -14,44 +14,68 @@ Description
+* pattern_max_sample_count: optional. The max sample logs to be returned per pattern in aggregation mode.
+* pattern_buffer_limit: optional. This is a special safeguard parameter for BRAIN algorithm to limit internal temporary buffer to hold processed logs.
+* new_field: The alias of the new field of the pattern. Default is "patterns_field".
+* algorithm parameters: optional. List of algorithm parameters. SIMPLE_PATTERN can specify regex by setting "pattern" parameter. BRAIN can specify variable_count_threshold(integer>0) and frequency_threshold_percentage(0.0f ~ 1.0f) parameters to tune variable token detection effect.
 
 Simple Pattern
 ============
-patterns [new_field=<new-field-name>] [pattern=<pattern>] <field>
+patterns <field> [by byClause...] [pattern_mode=LABEL | AGGREGATION] [pattern_max_sample_count=integer] [new_field=<new-field-name>] [pattern=<pattern>]
 
 or
 
-patterns [new_field=<new-field-name>] [pattern=<pattern>] <field> SIMPLE_PATTERN
+patterns <field> [by byClause...] pattern_method=SIMPLE_PATTERN [pattern_mode=LABEL | AGGREGATION] [pattern_max_sample_count=integer] [new_field=<new-field-name>] [pattern=<pattern>]
 
+* field: mandatory. The field must be a text field.
+* byClause: optional. The log groups to be labeled or aggregated. It could be fields and scalar functions.
+* pattern_method: optional. Specify pattern method to be simple_pattern. By default, it's simple_pattern if the setting ``plugins.ppl.default.pattern.method`` is not specified.
+* pattern_mode: optional. label mode or aggregation mode. Default is label mode.
+* pattern_max_sample_count: optional. The max sample logs to be returned per pattern in aggregation mode.
 * new-field-name: optional string. The name of the new field for extracted patterns, default is ``patterns_field``. If the name already exists, it will replace the original field.
 * pattern: optional string. The regex pattern of characters that should be filtered out from the text field. If absent, the default pattern is alphanumeric characters (``[a-zA-Z\d]``).
-* field: mandatory. The field must be a text field.
-* SIMPLE_PATTERN: Specify pattern method to be simple_pattern. By default, it's simple_pattern if the setting ``plugins.ppl.default.pattern.method`` is not specified.
 
 Brain
 ============
-patterns [new_field=<new-field-name>] [variable_count_threshold=<variable_count_threshold>] [frequency_threshold_percentage=<frequency_threshold_percentage>] <field> BRAIN
+patterns <field> pattern_method=BRAIN [new_field=<new-field-name>] [variable_count_threshold=<variable_count_threshold>] [frequency_threshold_percentage=<frequency_threshold_percentage>]
 
+* field: mandatory. The field must be a text field.
+* byClause: optional. The log groups to be labeled or aggregated. It could be fields and scalar functions.
+* pattern_method: optional. Specify pattern method to be brain. By default, it's simple_pattern if the setting ``plugins.ppl.default.pattern.method`` is not specified.
```

Comment:
> By default, it's simple_pattern if the setting ``plugins.ppl.default.pattern.method`` is not specified.

The default value is configured by the setting `plugins.ppl.default.pattern.method`.